scoutcloud/scoutcloud/src/logic/config/variables/rpc_url.rs

```diff
@@ -0,0 +1,143 @@
+use crate::logic::{
+    config::Error, ConfigValidationContext, ParsedVariable, ParsedVariableKey, UserVariable,
+};
+use anyhow::Context;
+use ethers::{prelude::*, providers::Provider, types::BlockNumber};
+use url::Url;
+
+pub struct RpcUrl(Url);
+
+#[async_trait::async_trait]
+impl UserVariable<Url> for RpcUrl {
+    fn new(v: Url, _config: &ConfigValidationContext) -> Result<Self, Error> {
+        Ok(Self(v))
+    }
+
+    async fn build_config_vars(
+        &self,
+        _config: &ConfigValidationContext,
+    ) -> Result<Vec<ParsedVariable>, Error> {
+        let mut parsed = vec![];
+
+        // check json rpc
+        let provider = Provider::<Http>::try_from(self.0.as_str())
+            .context("failed to parse url as http")
+            .map_err(|e| Error::Validation(e.to_string()))?;
+
+        check_jsonrpc_health(&provider)
+            .await
+            .map_err(|e| Error::Validation(e.to_string()))?;
+        parsed.push((
+            ParsedVariableKey::BackendEnv("ETHEREUM_JSONRPC_HTTP_URL".to_string()),
+            serde_json::Value::String(self.0.to_string()),
+        ));
+
+        // check trace method
+        // TODO: check trace method according to node_type
+        match check_any_trace_method(&provider).await {
+            Ok(()) => {
+                parsed.push((
+                    ParsedVariableKey::BackendEnv("ETHEREUM_JSONRPC_TRACE_URL".to_string()),
+                    serde_json::Value::String(self.0.to_string()),
+                ));
+                parsed.push((
+                    ParsedVariableKey::BackendEnv(
+                        "INDEXER_DISABLE_INTERNAL_TRANSACTIONS_FETCHER".to_string(),
+                    ),
+                    serde_json::Value::String("false".to_string()),
+                ));
+            }
+            Err(err) => {
+                tracing::warn!(
+                    err =? err,
+                    "`rpc_url` does not support tracing, disabling internal transactions"
+                );
+                parsed.push((
+                    ParsedVariableKey::BackendEnv(
+                        "INDEXER_DISABLE_INTERNAL_TRANSACTIONS_FETCHER".to_string(),
+                    ),
+                    serde_json::Value::String("true".to_string()),
+                ));
+            }
+        };
+
+        // check websocket
+        if let Some(ws_url) = get_any_healthy_ws_url(self.0.clone())
+            .await
+            .map_err(Error::Internal)?
+        {
+            parsed.push((
+                ParsedVariableKey::BackendEnv("ETHEREUM_JSONRPC_WS_URL".to_string()),
+                serde_json::Value::String(ws_url.to_string()),
+            ));
+        } else {
+            tracing::warn!(
+                "no valid websocket url found for `rpc_url`, skipping websocket configuration"
+            );
```

Comment:
Just to make sure, not setting websocket url does not require to set any disabling env, does it?

It seems that blockscout uses `ws://localhost:8546` as a default value in that case, so interested if that will not cause any problem for blockscout startup

Reply:
I just asked backend team, it's fine to simply omit WS env

about default `ws://localhost:8546`: I didn't find it in backend code. maybe it's just example value from docs?